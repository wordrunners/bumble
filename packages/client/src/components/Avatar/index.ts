export { Avatar } from './Avatar'
export type { User } from './Avatar'
