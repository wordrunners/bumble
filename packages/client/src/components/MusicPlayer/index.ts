export { MusicPlayer } from './MusicPlayer';
export { MusicData } from './MusicData';
