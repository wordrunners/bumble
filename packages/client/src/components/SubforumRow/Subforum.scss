.subforum-row {
  display: grid;
  grid-template-columns: 7% 60% 13% 20%;
}

.subforum-column {
  padding: 10px;
  margin: 4px;
  border-radius: 5px;
  background-color: var(--background2);
  & span {
    font-size: 0.8rem;
  }
}

.subforum-info {
  font-size: 0.8rem
}

.subforum-description * {
  margin-block: 0;
}

.center {
  display: flex;
  justify-content: center;
  align-items: center;
}

.subforum-icon .w-6-h-6 {
  font-size: 1rem;
  color: var(--word2);
}

h4 {
  & a {
    color: var(--word3);
  }
}

/* Для смартфонов */
@media screen and (max-width: 460px) {
  .subforum-row {
    display: grid;
    grid-template-columns: 20% 80%;
    grid-template-rows: 60% 40%;
  }

  .subforum-icon .w-6-h-6 {
    width: 2rem;
  }

  .subforum-column {
    padding: 3px;
    margin: 3px;
    border-radius: 5px;
    background-color: var(--background2);
    & small {
      font-size: 0.5rem;
    }
    & span {
      font-size: 0.5rem;
    }
  }

  .subforum-description {
    & p {
      font-size: 0.7rem;
    }
  }

  .subforum-info {
    font-size: 0.8rem
  }

}

/* Для планшетов */
@media screen and (min-width: 460px) and (max-width: 1024px) {


  .subforum-row {
    display: grid;
    grid-template-columns: 10% 57% 13% 20%;
  }

  .subforum-icon .w-6-h-6 {
    width: 2rem;
  }
}
