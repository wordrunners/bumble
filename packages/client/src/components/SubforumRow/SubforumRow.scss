.subforum-row {
  font-family: Arial, Helvetica, sans-serif;
  display: grid;
  grid-template-columns: 10% 65% 20%;
  min-width: 600px;
  gap: 8px;
  color: var(--text);

  &__wrapper {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
  }

  &__title {
    margin: 0 0 4px;
    word-wrap: break-word;
    font-size: 20px;
    line-height: 28px;
  }

  &__description {
    margin: 0;
    font-size: 14px;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    max-width: 250px;
  }

  &__date {
    margin: 0;
    font-size: 10px;
  }
}

.subforum-column {
  padding: 10px;
  margin-bottom: 8px;
  border-radius: 10px;
  background-color: var(--background2);
  & span {
    font-size: 0.8rem;
  }
}

.center {
  display: flex;
  justify-content: center;
  align-items: center;
}

.subforum-icon .w-6-h-6 {
  font-size: 1rem;
  color: var(--word2);
}

/* Для планшетов */
@media screen and (min-width: 460px) and (max-width: 1024px) {


  .subforum-row {
    display: grid;
    grid-template-columns: 10% 65% 20%;
    min-width: 400px;
  }

  .subforum-icon .w-6-h-6 {
    width: 2rem;
  }
}
