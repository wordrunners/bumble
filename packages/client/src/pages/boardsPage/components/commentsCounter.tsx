import { FC } from "react";

type Props = {
  open: boolean,
  counter: number
  onClick: () => void,
}

export const CommentsCounter: FC<Props> = ({open, onClick, counter}): JSX.Element => {
  return (
    <div className='comments-counter' onClick={onClick} style={{color: `blue`}}>
      <u>{!open ?  (!counter ? 'Ответить' : 'Ответы: ' + counter) : 'Скрыть'}</u>
    </div>
  )
}
