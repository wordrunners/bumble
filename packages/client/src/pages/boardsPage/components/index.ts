export { FormBox } from './formBox';
export { CommentBox } from './commentBox';
export { Like } from './like';
export { CommentsCounter } from './commentsCounter';
