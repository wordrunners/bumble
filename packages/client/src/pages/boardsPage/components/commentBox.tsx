import { FC, useState } from 'react'
import { addLike, Comment, selectBoardsData } from '@/store/boards'
import { useAuth, useAppDispatch, useAppSelector } from "@/hooks";
import { FormBox, Like } from '.'
import { Loader } from '@/components/Loader';
import '../boardPage.scss';
import { format } from "date-fns";

type Props = {
  like: boolean | undefined
  comment: Comment
  childComment: Comment[]
  userLogin: string;
}

export const CommentBox: FC<Props> = ({like, comment, childComment, userLogin}): JSX.Element => {
  const dispatch = useAppDispatch();

  const { user } = useAuth();
  const {  status } = useAppSelector(selectBoardsData)
  const formattedDate: string = format(new Date(comment?.createdAt), "dd-MM-yyyy, HH:mm");

  const clickLike = (value: boolean) => {
    if (user) {
      dispatch(
        addLike({
          isLike: value,
          commentId: comment.id,
          userId: user.id!,
          userLogin: user.login,
        })
      )
    }
  }

  const [open, setOpen] = useState(false)
  const handleOpen = () => {
    setOpen(previous => !previous)
  }

  return (
    <div className="forum__comment-box">
      {user && (
        <>
          <div className='forum__details'>
            <p className='forum__text forum__text_login'>{userLogin}</p>
            <p className="forum__date">{formattedDate}</p>
            <Like
              color={like ? '#dedc00' : 'grey'}
              onClick={() => clickLike(!like)}
            />
          </div>
          <p className='forum__text'>{comment.comment}</p>
          
          {!comment.parent_id && (
            <p onClick={handleOpen} className='forum__comments-counter'>
              {!open ?  (!childComment?.length ? 'Ответить' : 'Ответы: ' + childComment?.length) : 'Скрыть'}
            </p>
          )}

          {open && (
            status !== 'FETCH_FULFILLED' ? (
                <Loader />
              ) : (
                <>
                  {childComment?.length ? (
                    <div className='forum__answers'>
                      Ответы:
                      {childComment?.map(comment => {
                        return (
                          <div key={`comment-${comment.id}`}>
                            <p className='forum__text forum__text_login'>{comment.user_login}</p>
                            <p className='forum__text'>{comment.comment}</p>
                          </div>
                        )
                      })}
                    </div>
                  ) : (
                    null
                  )}
                  <FormBox
                    parentId={comment.id}
                    boardId={comment.board_id}
                  />
                </>
              )
          )}
        </>
      )}
    </div>
  )
}
