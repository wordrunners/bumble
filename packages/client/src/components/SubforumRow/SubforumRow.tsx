import './SubforumRow.scss';

type SubforumRowProps = {
  icon?: JSX.Element
  title: string
  description?: string
  numberOfComments?: string
  date?: string
}

export const SubforumRow = ({
  title, description, icon, numberOfComments, date,
}: SubforumRowProps): JSX.Element => {
  return (
    <div className="subforum-row">
      <div className="subforum-icon subforum-column center">{icon}</div>
      <div className="subforum-description subforum-column">
        <h4 className="subforum-row__title">{title}</h4>
        <div className='subforum-row__wrapper'>
          <p className="subforum-row__description">{description}</p>
          <p className="subforum-row__date">{date}</p>
        </div>
      </div>
      <div className="subforum-stats subforum-column center">
        <span>{numberOfComments}</span>
      </div>
    </div>
  )
}
