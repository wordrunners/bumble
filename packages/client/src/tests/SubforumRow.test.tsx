import { SubforumRow } from '../components/SubforumRow'
import { render } from '@testing-library/react'

test('SubforumRow', async () => {
  const subforumRow = render(
    <SubforumRow title={"This is SubforumRow"}></SubforumRow>
  );
  expect(subforumRow).toMatchSnapshot();
}); 
