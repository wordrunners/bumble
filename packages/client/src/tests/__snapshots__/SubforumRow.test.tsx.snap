// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubforumRow 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="subforum-row"
      >
        <div
          class="subforum-icon subforum-column center"
        />
        <div
          class="subforum-description subforum-column"
        >
          <h4>
            <a
              href="#"
            >
              This is SubforumRow
            </a>
          </h4>
          <p />
        </div>
        <div
          class="subforum-stats subforum-column center"
        >
          <span>
             | 
          </span>
        </div>
        <div
          class="subforum-info subforum-column"
        >
          <b>
            <a
              href=""
            />
          </b>
           
          от 
          <a
            href=""
          />
           
          <br />
          <small />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="subforum-row"
    >
      <div
        class="subforum-icon subforum-column center"
      />
      <div
        class="subforum-description subforum-column"
      >
        <h4>
          <a
            href="#"
          >
            This is SubforumRow
          </a>
        </h4>
        <p />
      </div>
      <div
        class="subforum-stats subforum-column center"
      >
        <span>
           | 
        </span>
      </div>
      <div
        class="subforum-info subforum-column"
      >
        <b>
          <a
            href=""
          />
        </b>
         
        от 
        <a
          href=""
        />
         
        <br />
        <small />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
